projects/experimental/components/avatar/avatar.component.ts

```diff
@@ -17,18 +19,24 @@ export class TuiAvatarComponent {
     round = false;
 
     @Input()
-    src: string | null = null;
+    src: SafeResourceUrl | string | null = null;
 
     @HostBinding('class._img')
     get img(): boolean {
         return this.type === 'img';
     }
 
-    get value(): string {
+    get value(): SafeResourceUrl | string {
         return this.src || '';
     }
 
     get type(): 'content' | 'icon' | 'img' | 'text' {
+        if (!tuiIsString(this.value)) {
```

Comment:
I think just if it is not a string and not null — return img

Reply:
fixed